### ISSUE TITLE
ngAnimate: not performing enter animation in ng-repeat when using ng-class (with classNameFilter)

### ISSUE THREAD

**Issue by MikiSoft** [angular/angular.js#14915]
Other animations work, but not the entering one.  
Example: http://plnkr.co/edit/NVc4bc3kSj5K0WlbVBex?p=preview  
**Update:** http://plnkr.co/edit/jtPQ6BL2uhZ9kZapft8E?p=preview - here's another example using `uib-tab` directive; it won't work on the first tab change event (i.e. when it's calling `enableAnimation()` for the first time), so it seems that it's because when calling `enabled` function it acts asynchronously?

**Comment by gkalpak** [comment]
This is because `ngClass` will trigger an animation of its own: ann `addClass` animation. [As documented][1], the class will be added asynchronously, which means that it won't be present when the `enter` animation will be looking for it.
You can use `class+interpolation` as an alternative way of applying the class immediately: [Updated plnkr][2]

[1]: https://docs.angularjs.org/api/ng/service/$animate#addClass
[2]: http://plnkr.co/edit/uT9ZWA9NHcl5ql1SC7uk?p=preview

**Comment by MikiSoft** [comment]
Thanks @gkalpak for explanation and assistance. What about the second example which I posted above?

**Comment by gkalpak** [comment]
Actually, my work-around is invalid (it ony works by chance in the particular usecase).
Further investigation is needed about the 2nd demo - there might be an issue with `ngAnimate`.

**Comment by Narretz** [comment]
Isn't this plnkr NOT supposed to work: http://plnkr.co/edit/7fOdQw7Q02WwzeZlfgTu?p=preview

All animations work for me.

**Comment by MikiSoft** [comment]
@Narretz yes they work but what I want is to disable animation of the initial elements.

**Comment by gkalpak** [comment]
http://plnkr.co/edit/7fOdQw7Q02WwzeZlfgTu?p=preview works "by chance" (see https://github.com/angular/angular.js/issues/14915#issuecomment-233146200). But that's just a problem with my faulty work-around.

The real problem is with http://plnkr.co/edit/jtPQ6BL2uhZ9kZapft8E?p=preview. From a quick look I had the other day, it seems that even when animations are correctly detected as disabled, the `addClass`-animation related classes (e.g. `<my-class>-add` and `<my-class>-add-active`) are still applied, which causes the elements to animate. I suspect there might be a bug in `ngAnimate`.
<sub>(Take this with a couple of grains of salt, because it was a really brief look :stuck_out_tongue: </sub>

I plan to take a proper look at some point, unless someone beats me to it :wink:

**Comment by MikiSoft** [comment]
@gkalpak What about the other example for which I've talked in my previous comment?

**Comment by gkalpak** [comment]
@MikiSoft , we first need to fix any bugs that affect the usecase and then see if/how it is possible (and possibly make it...possible, if it isn't already).
